src/lib/dma.c

```diff
@@ -143,6 +151,9 @@ void dma_put(struct dma *dma)
 	}
 	trace_event(TRACE_CLASS_DMA, "dma_put(), dma = %p, sref = %d",
 		   (uintptr_t)dma, dma->sref);
+
+	dcache_writeback_region(dma, sizeof(*dma));
```

Comment:
I think your concept is, that the `lib_dma` array can only be modified after a `dma_get()` and only on the same single core. So, it's sufficient to just write back caches during a `dma_put()`. But then wouldn't it be better to then also just invalidate the cache for this single object immediately here, then we wouldn't need to invalidate all array members in `dma_get()`?